docs/samples/custom/grpc-server/README.md

```diff
@@ -0,0 +1,94 @@
+# A sample for gRPC Server in custom model
+
+This sample can be used to try out gRPC, HTTP/2, and custom port configuration
+in a custom inference  service.
+
+## Build and Deploy the sample code
+
+1. Use Docker to build a container image for this service and push to Docker Hub.
+
+  Replace `{username}` with your Docker Hub username then run the commands:
+
+  ```shell
+  # Build the container on your local machine.
+  docker build -t {username}/helloworld-grpc .
+
+  # Push the container to docker registry.
+  docker push {username}/helloworld-grpc
+  ```
+
+3. Update the `grpc-service.yaml` file in the project to reference the published image from step 1.
+
+   Replace `{username}` in `grpc-service.yaml` with your Docker Hub user name:
+   
+   
+  ```yaml
+ apiVersion: serving.kubeflow.org/v1alpha2
+kind: InferenceService
+metadata:
+  labels:
+    controller-tools.k8s.io: "1.0"
+  name: grpc-service
+spec:
+  default:
+    predictor:
+      custom:
+        container:
+          image: docker.io/{username}/helloworld-grpc:latest
+          ports:
+          - name: h2c
+            containerPort: 8080
+  ```
+
+4. Use `kubectl` to deploy the service.
+
+  ```shell
+  kubectl apply -f grpc-service.yaml
+  ```
+
+## Testing the service
+
+**Get the inference service address:**
+
+```
+kubectl get ksvc grpc-service-predictor-default
+```
+
+**Expect output**
+
+```
+NAME                             URL                                                                  READY   REASON
+grpc-service-predictor-default   http://grpc-service-predictor-default.default.1.2.3.4.xip.io   True    
+```
+**How to setup the magic dns**
+[Install Istio for Knative](https://knative.dev/docs/install/installing-istio/)
```

Comment:
Can you mention that it is the `Configure DNS` section and also link this sample here https://github.com/kubeflow/kfserving/tree/master/docs/samples#deploy-kfserving-inferenceservice-with-a-custom-predictor ? thanks!

Reply:
OK